src/gr11review/part2/Utility.java

```diff
@@ -1,9 +1,352 @@
+            //When counter equals array length return true
+            else if (inner[innerIndex] == outer[counter]){
+
+                innerIndex++;
+                counter = 0;
+
+                if (innerIndex == inner.length){
+
+                    return true;
+
+                }
+
+            }
+
+            
+
+        }
+
+        return false;
+
+      }
+  
+      /**
+        * Array 4
+        * Returns true if an integer array can be split so that the sum of the numbers on both sides are equal
+        * @author P. Tran
+        */
+      public static boolean canBalance(int[] nums) {
+        int arrayLength = nums.length;
+
+        int arraySum = 0;
+        int leftSide = 0;
+
+        //Calculate sum of array
+        for (int i = 0; i < arrayLength; i++) {
+          arraySum += nums[i];
+        }
+      
+        //If leftSide of array is equal to half of arraySum at any point, array can be split
+        for (int j = 0; j < arrayLength; j++) {
+          leftSide += nums[j];
+            
+          if (leftSide * 2 == arraySum) {
+            return true;
+          }
+        }
+
+        return false;
+	    }
+
+        /**
+          * Array 5 - Two Dimensional
+          * Prints a pascal's traingle with a size based on the inputed numbers
+          * @author A. Wong
+          */
+
+        public static void pascalTri(int i, int j) throws IOException{
+
+        //Sets up an array
+        int[][] pascalArr = new int[i][j];
+
+        //Sets up a file printer
+        File myObj = new File("pascalOut.txt");
+        FileWriter myWriter = new FileWriter("pascalOut.txt");
+        
+        myObj.createNewFile();
+
+        for (int rowIndex = 0; rowIndex < i; rowIndex++){
+
+            for (int columnIndex = 0; columnIndex < j; columnIndex++){
+
+                //If the row or index is 0 set number to 1
+                if (rowIndex == 0 || columnIndex == 0){
+
+                    pascalArr[rowIndex][columnIndex] = 1;
+
+                }
+                //Adds the number above and the number to the left
+                else{
+
+                    pascalArr[rowIndex][columnIndex] = pascalArr[rowIndex - 1][columnIndex] + pascalArr[rowIndex][columnIndex - 1]; 
+
+                }
+
+                //If it is not the right most column, include a comma
+                if (columnIndex < (j - 1)){
+
+                    myWriter.write(pascalArr[rowIndex][columnIndex] + ",");
+
+                }
+
+            }
+
+            //If it is the right most column make a new line
+            myWriter.write(pascalArr[rowIndex][j - 1] + "\n");
+            
+
+        }
+
+        myWriter.close();
+
+       }
+  
+  /**
+     * Array 6
+     * Prints onto a text file a two dimensional array of size n
+     * @author P. Tran
+     */
+  public static void diagonal(int n)throws IOException {
+
+    try {
+      PrintWriter theout = new PrintWriter(new FileWriter("diagonalOut.txt", true));
```

Comment:
when you create your out file, you don't want to set the last parameter to `true` since this will append to the contents of the file if the file already exists.  set it to false or don't include a value (the default is false)
```
PrintWriter theout = new PrintWriter(new FileWriter("diagonalOut.txt", true));
```

https://github.com/SACHSTech/native-java-and-grade-11-review-assignment---part-2-adrian-w-and-paul-t/runs/2625916165#step:3:186